packages/react-native/tests/storage/FileBreadcrumbsStorage.spec.ts

```diff
@@ -0,0 +1,415 @@
+        assert(mainStream);
+
+        const actualMain = await loadBreadcrumbs(fs, mainStream);
+        expect(actualMain).toEqual(expectedMain);
+    });
+
+    it('should return added breadcrumbs in two attachments', async () => {
+        const fs = mockStreamFileSystem();
+        const session = new SessionFiles(fs, '.', 'sessionId');
+
+        const breadcrumbs: RawBreadcrumb[] = [
+            {
+                level: BreadcrumbLogLevel.Info,
+                message: 'a',
+                type: BreadcrumbType.Manual,
+                attributes: {
+                    foo: 'bar',
+                },
+            },
+            {
+                level: BreadcrumbLogLevel.Debug,
+                message: 'b',
+                type: BreadcrumbType.Http,
+            },
+            {
+                level: BreadcrumbLogLevel.Warning,
+                message: 'c',
+                type: BreadcrumbType.Navigation,
+                attributes: {},
+            },
+        ];
+
+        const expectedMain: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'warning',
+                message: 'c',
+                timestamp: expect.any(Number),
+                type: 'navigation',
+                attributes: {},
+            },
+        ];
+
+        const expectedFallback: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'info',
+                message: 'a',
+                timestamp: expect.any(Number),
+                type: 'manual',
+                attributes: {
+                    foo: 'bar',
+                },
+            },
+            {
+                id: expect.any(Number),
+                level: 'debug',
+                message: 'b',
+                timestamp: expect.any(Number),
+                type: 'http',
+            },
+        ];
+
+        const storage = new FileBreadcrumbsStorage(session, fs, {
+            maximumBreadcrumbs: 4,
+        });
+
+        for (const breadcrumb of breadcrumbs) {
+            storage.add(breadcrumb);
+            await nextTick();
+        }
+
+        // FileBreadcrumbsStorage is asynchronous in nature
+        await nextTick();
+
+        const [fallbackAttachment, mainAttachment] = storage.getAttachments();
+
+        const mainStream = mainAttachment.get();
+        const fallbackStream = fallbackAttachment.get();
+        assert(mainStream);
+        assert(fallbackStream);
+
+        const actualMain = await loadBreadcrumbs(fs, mainStream);
+        const actualFallback = await loadBreadcrumbs(fs, fallbackStream);
+        expect(actualMain).toEqual(expectedMain);
+        expect(actualFallback).toEqual(expectedFallback);
+    });
+
+    it('should return no more than maximumBreadcrumbs breadcrumbs', async () => {
+        const fs = mockStreamFileSystem();
+        const session = new SessionFiles(fs, '.', 'sessionId');
+
+        const breadcrumbs: RawBreadcrumb[] = [
```

Comment:
small nit: 
this file is quite long and can be simplified by moving breadcrumbs outside each test to a describe scope. You're always creating the same breadcrumbs so .. why not?

Reply:
Why do we care about file length for tests? It's also easier to modify a test if the value is inside each case.